### PR TITLE
Feature/add get raw uri

### PR DESCRIPTION
**Pull request by bgsavage**
Added the `getRawUri` method to ArchiveService and related tests.

### CODE REVIEW

File: archive/src/test/java/org/opentestsystem/rdw/archive/LocalArchiveServiceTest.java [hunk 1]
```diff
@@ -142,4 +142,25 @@ public void itShouldGuessContentTypeSometimes() {
         assertThat(LocalArchiveService.guessContentType("test.xls")).isNull();
         assertThat(LocalArchiveService.guessContentType("file:///test.txt")).isEqualTo("text/plain");
```
Comment (bgsavage) [inline archive/src/test/java/org/opentestsystem/rdw/archive/LocalArchiveServiceTest.java]: Thanks @malaffoon for adding these tests.

File: archive/src/main/java/org/opentestsystem/rdw/archive/S3ArchiveService.java [hunk 2]
```diff
@@ -115,9 +120,9 @@ public void writeResource(final String location, final InputStream is, final Pro
         try {
             amazonS3.putObject(bucket, location, content, metadata);
         } catch (final AmazonServiceException e) {
-            final String msg = "Error writing content to " + bucket + "/" + location;
-            logger.warn(msg, e);
-            throw new RuntimeException(msg, e);
+            final String message = "Error writing content to " + getRawUri(location);
+            logger.warn(message, e);
+            throw new RuntimeException(message, e);
```
Comment (bgsavage) [inline archive/src/main/java/org/opentestsystem/rdw/archive/S3ArchiveService.java]: unrelated rename.

File: archive/src/main/java/org/opentestsystem/rdw/archive/ArchiveService.java [hunk 3]
```diff
@@ -16,6 +16,13 @@
     String RawURI = "rawURI";
 
     /**
+     * Gets the Archive Service fully qualified URI for a given location.
+     * @param location unique (relative) location for content, e.g. "EXAM/4F/0A/4F0A1245"
```
Comment (malaffoon) [inline archive/src/main/java/org/opentestsystem/rdw/archive/ArchiveService.java]: Perhaps add a clause like `regardless of whether a resource exists for the given location`.

File: archive/src/main/java/org/opentestsystem/rdw/archive/LocalArchiveService.java [hunk 4]
```diff
@@ -55,6 +55,11 @@ public LocalArchiveService(final String root) {
     }
 
     @Override
+    public String getRawUri(final String location) {
+        return FileSchemePrefix + rootPath + location;
```
Comment (malaffoon) [inline archive/src/main/java/org/opentestsystem/rdw/archive/LocalArchiveService.java]: I think you might want to go through the realResourceFile() method. All that that fiddling with getCanonicalFile is there because of symlinks and stuff. I think the most correct form of this might be: `return FileSchemePrefix + realResourceFile(location).getAbsolutePath();`

File: archive/src/test/java/org/opentestsystem/rdw/archive/LocalArchiveServiceTest.java [hunk 5]
```diff
@@ -142,4 +142,25 @@ public void itShouldGuessContentTypeSometimes() {
...
     }
+    
+    @Test
+    public void itShouldReturnUriForALocation(){
+        String location = "some/location/file.pdf";
```
Comment (malaffoon) [inline archive/src/test/java/org/opentestsystem/rdw/archive/LocalArchiveServiceTest.java]: nit: `final`

File: archive/src/main/java/org/opentestsystem/rdw/archive/ArchiveService.java (hunk 3 above)
Comment (bgsavage) [reply archive/src/main/java/org/opentestsystem/rdw/archive/ArchiveService.java]: 👍

File: archive/src/main/java/org/opentestsystem/rdw/archive/LocalArchiveService.java (hunk 4 above)
Comment (bgsavage) [reply archive/src/main/java/org/opentestsystem/rdw/archive/LocalArchiveService.java]: Hm, so we want to convert symlinks to absolute paths?  I'm not doing this with the export-root in ArchiveExamExportService... If we should be doing it there, than that's only specific to a local implementation, which would warrant separate implementations again.
